ts/main.ts

```diff
@@ -1 +1,51 @@
 /* global data */
+
+interface FormElements extends HTMLFormControlsCollection {
+  photoUrl: HTMLInputElement;
+  title: HTMLInputElement;
+  notes: HTMLTextAreaElement;
+}
+
+const $image = document.querySelector('#img') as HTMLImageElement;
+const $newImg = document.getElementById('photo-url') as HTMLInputElement;
+const $form = document.querySelector('#form') as HTMLFormElement;
+function imgChanger(event: Event): void {
+  const $eventTarget = event.target as HTMLInputElement;
+  const $eventTargetURL = $eventTarget.value;
+  if (urlChecker($eventTargetURL)) {
+    $image.src = $eventTargetURL;
+  } else {
+    $image.src =
+      'https://www.pokemon.com/static-assets/content-assets/cms2/img/pokedex/full/403.png';
+  }
+}
+
+function urlChecker(string: string): boolean {
+  const res = string.match(
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g,
+  );
+  return res !== null;
+}
+
+if (!$newImg) throw new Error('The Image did not load.');
+if ($newImg) $newImg.addEventListener('change', imgChanger);
+
```

Comment:
You move this up correctly, but on line 31 you should remove the `if ($newImg)` because your line 30 is already guaranteeing that `$newImg` is truthy